.github/workflows/set-milestone-on-pr.yml

```diff
@@ -0,0 +1,63 @@
+#
+# Copyright (c) 2020 Red Hat, Inc.  and others.
+# This program and the accompanying materials are made available under the
+# terms of the Eclipse Public License v. 2.0 which is available at
+# http://www.eclipse.org/legal/epl-2.0.
+#
+# This Source Code may also be made available under the following Secondary
+# Licenses when the conditions for such availability set forth in the Eclipse
+# Public License v. 2.0 are satisfied: GNU General Public License, version 2
+# with the GNU Classpath Exception which is available at
+# https://www.gnu.org/software/classpath/license.html.
+#
+# SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+
+# Set milestone on each pull request by using the next minor version
+# next version is computed using npm version tool
+on:
+  pull_request:
+    branches: [master]
+    types: [closed]
+
+jobs:
+  set-milestone:
+    if: github.event.pull_request.merged == true
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - id: compute-milestone
+        run: |
+          cd packages/core
+          echo "MILESTONE_NUMBER=$(npm --no-git-tag-version version minor | cut -c 2-)" >> $GITHUB_ENV
+      - id: set
+        uses: actions/github-script@v3
+        with:
+          github-token: ${{secrets.GITHUB_TOKEN}}
+          script: |
+            // get milestone
+            const milestone = process.env.MILESTONE_NUMBER
+            console.log(`Using milestone with name ${milestone}`)
```

Comment:
Just to better understand the process: With what you have, when we publish `1.7.0` it will create a `1.7.0` milestone and add all following merged PR into that milestone. So if someone wants to know what was part of `1.7.0` as they pull it from npm, they'd need to look at the `1.6.0` milestone? Is this how it should work?

Reply:
it's assigning milestones to pull requests.

So let say I merge a Pull Request today, we grab current version + (minor+1) which is 1.7.0
so we'll set milestone on this merged PR to `1.7.0`

when publishing `1.7.0`, version that will be set on PRs will be `1.8.0`

so for people going to https://github.com/eclipse-theia/theia/milestones they will find automatically what has been added into 1.7.0, 1.8.0, etc.

Reply:
Got it, perfect!